### PR TITLE
Add availability_domain field to GCE instance scheduling

### PR DESCRIPTION
**Pull request by kubalaguna**
**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
compute: added `availability_domain` field to `google_compute_instance` resource
```

### CODE REVIEW

**Review by rileykarson — DISMISSED** [review]
LGTM assuming tests pass! (will merge after results if so)

File: mmv1/third_party/terraform/services/compute/compute_instance_helpers.go.erb
Comment (rileykarson) [inline mmv1/third_party/terraform/services/compute/compute_instance_helpers.go.erb:278]: We're writing `availability_domain` to instance template due to the shared code, but the value isn't available. I assume the unset value is `nil` so we can't just use a conditional write, an approach used in some other places. We may want to add the field to schema in instance template and add a note it doesn't currently work (but that could lead to weird behaviour if it's added in the future).

Otherwise we may need to pass a flag through from the initial callsite about whether we're working on a template or actual instance, and use that to control what we write.

(I would personally like to remove the shared code eventually to avoid these kind of decisions, not that maintaining separate copies will be very fun 🙁 )

**Comment by c2thorn** [comment]
Hi @kubalaguna 

Tomorrow, the Magic Modules repository is scheduled to undergo a language migration from Ruby to Go. You can view more details about this in our announcement here: https://github.com/hashicorp/terraform-provider-google/issues/19583 (or [go/mm-migration-announcement](go/mm-migration-announcement) if you are a Googler)

This open pull request may become incompatible due to most YAML and .erb files converting to Go-compatible files.

Our team (Magic Modules repository maintainers) has tooling to automatically convert changes to the new language, and we can prepare a new commit for this pull request that is compatible with the migration.

In order to push the new changes to your pull request, we would need to <ins>force push the commit to your fork's branch</ins>. Our tooling saves a backup branch before converting, so we could rollback or open a new pull request if needed. We would also work with you and the PR reviewer in the event additional changes are needed.

You also have the option to update the pull request yourself after the migration. You can view a [preview branch](https://github.com/GoogleCloudPlatform/magic-modules/tree/FEATURE-BRANCH-rewrite-switchover-preview) and [updated documentation](https://googlecloudplatform.github.io/magic-modules/reference/ruby-go-changes/) related to the migration changes.

We will take no action until we have your explicit permission to push changes to your fork's branch used for this pull request. Let me or your reviewer know if you have any further questions!

**Comment by c2thorn** [comment]
It has been several weeks and this PR is likely too far out of date to convert with our tooling. Please disregard my previous comment and recreate this PR in the new format.

**Pull request closed without merging**
